File: app/src/main/java/ch/bailu/aat/gpx/GpxDataWrapper.java
package ch.bailu.aat.gpx;


import ch.bailu.aat.coordinates.BoundingBoxE6;
import ch.bailu.aat.gpx.interfaces.GpxBigDeltaInterface;
import ch.bailu.aat.gpx.interfaces.GpxDeltaPointInterface;
import ch.bailu.aat.gpx.interfaces.GpxPointInterface;
import ch.bailu.aat.gpx.interfaces.GpxType;


public class GpxDataWrapper implements GpxDeltaPointInterface, GpxBigDeltaInterface {

    private GpxBigDeltaInterface summary;
    private GpxPointInterface point;

    public GpxDataWrapper() {
        setVisibleTrackSegment(GpxBigDelta.NULL);
        setVisibleTrackPoint(GpxPoint.NULL);
    }

    public void setVisibleTrackSegment(GpxBigDeltaInterface s) {
        summary = s;
    }

    public void setVisibleTrackPoint(GpxPointInterface p) {
        point = p;
    }


    @Override
    public int getLatitudeE6() {
        return point.getLatitudeE6();
    }

    @Override
    public int getLongitudeE6() {
        return point.getLongitudeE6();
    }
    
    @Override
    public float getSpeed() {
        return summary.getSpeed();
    }

    @Override
    public float getDistance() {
        return summary.getDistance();
    }

    @Override
    public long getStartTime() {
        return summary.getStartTime();
    }

    @Override
    public long getTimeDelta() {
        return summary.getTimeDelta();
    }


    @Override
    public long getPause()  {
        return summary.getPause();
    }

    @Override
    public short getAltitude() {
        return point.getAltitude();
    }

    @Override
    public double getLatitude() {
        return point.getLatitude();
    }

    @Override
    public double getLongitude() {
        return point.getLongitude();
    }

    @Override
    public long getTimeStamp() {
        return point.getTimeStamp();
    }

    @Override
    public float getAcceleration() {
        return summary.getAcceleration();
    }


    @Override
    public BoundingBoxE6 getBoundingBox() {
        return summary.getBoundingBox();
    }

    @Override
    public long getEndTime() {
        return summary.getEndTime();
    }



    @Override
    public GpxType getType() {
        return summary.getType();
    }

    @Override
    public GpxAttributes getAttributes() {
        return point.getAttributes();
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/GpxDataWrapper.kt
package ch.bailu.aat_lib.gpx

import ch.bailu.aat_lib.coordinates.BoundingBoxE6
import ch.bailu.aat_lib.gpx.attributes.GpxAttributes
import ch.bailu.aat_lib.gpx.interfaces.GpxBigDeltaInterface
import ch.bailu.aat_lib.gpx.interfaces.GpxDeltaPointInterface
import ch.bailu.aat_lib.gpx.interfaces.GpxPointInterface
import ch.bailu.aat_lib.gpx.interfaces.GpxType

open class GpxDataWrapper : GpxDeltaPointInterface, GpxBigDeltaInterface {
    private var summary: GpxBigDeltaInterface = GpxBigDelta.NULL
    private var point: GpxPointInterface = GpxPoint.NULL

    fun setVisibleTrackSegment(s: GpxBigDeltaInterface) {
        summary = s
    }

    fun setVisibleTrackPoint(p: GpxPointInterface) {
        point = p
    }

    override fun getLatitudeE6(): Int {
        return point.getLatitudeE6()
    }

    override fun getLongitudeE6(): Int {
        return point.getLongitudeE6()
    }

    override fun getSpeed(): Float {
        return summary.getSpeed()
    }

    override fun getDistance(): Float {
        return summary.getDistance()
    }

    override fun getStartTime(): Long {
        return summary.getStartTime()
    }

    override fun getTimeDelta(): Long {
        return summary.getTimeDelta()
    }

    override fun getPause(): Long {
        return summary.getPause()
    }

    override fun getAltitude(): Float {
        return point.getAltitude()
    }

    override fun getLatitude(): Double {
        return point.getLatitude()
    }

    override fun getLongitude(): Double {
        return point.getLongitude()
    }

    override fun getTimeStamp(): Long {
        return point.getTimeStamp()
    }

    override fun getAcceleration(): Float {
        return summary.getAcceleration()
    }

    override fun getBoundingBox(): BoundingBoxE6 {
        return summary.getBoundingBox()
    }

    override fun getEndTime(): Long {
        return summary.getEndTime()
    }

    override fun getType(): GpxType {
        return summary.getType()
    }

    override fun getAttributes(): GpxAttributes {
        return summary.getAttributes()
    }
}
